### ISSUE TITLE
Format document breaks when using multi-line functions

### ISSUE THREAD

**Issue by brwilkinson** [microsoft/vscode-azurearmtools#435]
**Version 0.8.3 (2019-12-16)**

When taking advantage of the new multi line capabilities for functions, it breaks formatting.

E.g.

I deploy this and it deploys just fine

![image](https://user-images.githubusercontent.com/3605266/71388819-c94c9d00-25ae-11ea-8c05-c0377fc32ccf.png)

Now if you format the document or in this case for the example format select

![image](https://user-images.githubusercontent.com/3605266/71388852-ef723d00-25ae-11ea-8711-c561d7f03106.png)

It's now broken.

If the resourceid function is all on one line, formatting works as expected

e.g.

![image](https://user-images.githubusercontent.com/3605266/71388915-2cd6ca80-25af-11ea-9627-89ac655d096b.png)

**Comment by StephenWeatherford** [comment]
Thanks for reporting!  I'm guessing if the ending ] and " are placed together on the same line that if would deploy, will check later.

To help us decide how to deal with this, what would be your expectation when formatting multi-line expressions?  Do you want them broken out into separate lines?  Do you want them not touched at all?  Thanks!

**Comment by brwilkinson** [comment]
@StephenWeatherford 

**Initial preferece**
I would say not touch them at all... since I likely (manually) break them out into the different segments of the resource provider Etc. to enhance readability.

However . . .

**Secondary reaction**
If that was assumed, then I can't see a clear path to somehow have a auto formatting setting that may break down long expressions... to enhance readability.

I think fixing this current bug with the first option would certainly be a huge win.... not sure how easy the second part would be anyway?

**Comment by StephenWeatherford** [comment]
Makes sense, thanks.

**Comment by brwilkinson** [comment]
here i another sample of how I think I would like to break it out for longer one... in case it helps with option 2

![image](https://user-images.githubusercontent.com/3605266/71399560-de3e2600-25d8-11ea-9f25-f9b8efcb8c09.png)

**Comment by nilshedstrom** [comment]
Before sorting:
```
{
    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        //Comment before parameter 2
        "parameter1": {
            // Comment inside parameter 1
           "type": "string",
           "metadata": {
                "description": "description"
            }
        },
        //Comment before parameter 1
        "parameter2": {
            // Comment inside parameter 2
           "type": "string",
           "metadata": {
                "description": "description"
            }
        }
    },
    "variables": {},
    "resources": [{
        "name": "storageaccount2",
        "type": "Microsoft.Storage/storageAccounts",
        "apiVersion": "2015-06-15",
        "location": "[resourceGroup().location]",
        "tags": {
            "displayName": "storageaccount1"
        },
        "properties": {
            "accountType": "Standard_LRS"
        },
        "dependsOn": [
            "[resourceId('Microsoft.DocumentDb/databaseAccounts/sqlDatabases/containers',
            toLower('instanceName'),
            toLower('databaseName'),
            toLower('containerName'))]"
        ]
    },
    {
        "name": "storageaccount1",
        "type": "Microsoft.Storage/storageAccounts",
        "apiVersion": "2015-06-15",
        "location": "[resourceGroup().location]",
        "tags": {
            "displayName": "storageaccount1"
        },
        "properties": {
            "accountType": "Standard_LRS"
        }
    }],
    "outputs": {},
    "functions": []
}
```
After sorting:
```
{
    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        //Comment before parameter 2
        "parameter1": {
            // Comment inside parameter 1
           "type": "string",
           "metadata": {
                "description": "description"
            }
        },
        //Comment before parameter 1
        "parameter2": {
            // Comment inside parameter 2
           "type": "string",
           "metadata": {
                "description": "description"
            }
        }
    },
    "variables": {},
    "resources": [{
        "name": "storageaccount1",
        "type": "Microsoft.Storage/storageAccounts",
        "apiVersion": "2015-06-15",
        "location": "[resourceGroup().location]",
        "tags": {
            "displayName": "storageaccount1"
        },
        "properties": {
            "accountType": "Standard_LRS"
        }
    },
    {
        "name": "storageaccount2",
        "type": "Microsoft.Storage/storageAccounts",
        "apiVersion": "2015-06-15",
        "location": "[resourceGroup().location]",
        "tags": {
            "displayName": "storageaccount1"
        },
        "properties": {
            "accountType": "Standard_LRS"
        },
        "dependsOn": [
            "[resourceId('Microsoft.DocumentDb/databaseAccounts/sqlDatabases/containers',
            toLower('instanceName'),
            toLower('databaseName'),
            toLower('containerName'))]"
        ]
    }],
    "outputs": {},
    "functions": []
}
```
Conclusions
- Sorting works good when the comments are inside the parameter for instance
- Sorting does not work when the comment are outside the parameter for instance
- Sorting seems to work fine with multi line strings

**Comment by StephenWeatherford** [comment]
@nilshedstrom I believe you added this to the wrong issue?  Thx.

**Comment by StephenWeatherford** [comment]
Tracked here: https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1049548

**Comment by ishepherd** [comment]
+1, encountered this when refactoring a giant messy ARM template.

**Comment by StephenWeatherford** [comment]
We have a fix...

**Comment by StephenWeatherford** [comment]
Need to add functional test.

**Comment by StephenWeatherford** [comment]
Fix released with 0.11.0
